Apps/Admin/Client/Store/Dashboard/DashboardActions.cs

```diff
@@ -87,64 +97,110 @@ public LoadDependentsAction(int timeOffset)
-        /// <param name="startPeriod">The period start over which to evaluate the user.</param>
-        /// <param name="endPeriod">The period end over which to evaluate the user.</param>
-        /// <param name="timeOffset">The offset from the client browser to UTC.</param>
-        public LoadRecurringUsersAction(int days, string startPeriod, string endPeriod, int timeOffset)
+        /// <param name="data">user data.</param>
+        public LoggedInSuccessAction(IDictionary<DateTime, int> data)
+            : base(data)
         {
-            this.Days = days;
-            this.StartPeriod = startPeriod;
-            this.EndPeriod = endPeriod;
-            this.TimeOffset = timeOffset;
         }
+    }
 
+    /// <summary>
+    /// The action representing a failed logged in user action.
+    /// </summary>
+    public class LoggedInUserFailAction : BaseFailAction
+    {
         /// <summary>
-        /// Gets or sets days.
+        /// Initializes a new instance of the <see cref="LoggedInUserFailAction"/> class.
         /// </summary>
-        public int Days { get; set; }
+        /// <param name="error">The request error.</param>
+        public LoggedInUserFailAction(RequestError error)
+            : base(error)
+        {
+        }
+    }
 
+    /// <summary>
+    /// The action representing the initiation of dependents load action.
+    /// </summary>
+    public class DependentsAction
+    {
         /// <summary>
-        /// Gets or sets start period.
+        /// Initializes a new instance of the <see cref="DependentsAction"/> class.
         /// </summary>
-        public string StartPeriod { get; set; }
+        /// <param name="timeOffset">The offset from the client browser to UTC.</param>
+        public DependentsAction(int timeOffset)
+        {
+            this.TimeOffset = timeOffset;
+        }
 
         /// <summary>
-        /// Gets or sets end period.
+        /// Gets or sets time offset.
         /// </summary>
-        public string EndPeriod { get; set; }
+        public int TimeOffset { get; set; }
+    }
 
+    /// <summary>
+    /// The action representing a successful dependents action.
+    /// </summary>
+    public class DependentsSuccessAction : BaseSuccessAction<IDictionary<DateTime, int>>
+    {
         /// <summary>
-        /// Gets or sets time offset.
+        /// Initializes a new instance of the <see cref="DependentsSuccessAction"/> class.
         /// </summary>
-        public int TimeOffset { get; set; }
+        /// <param name="data">user data.</param>
+        public DependentsSuccessAction(IDictionary<DateTime, int> data)
+            : base(data)
+        {
+        }
     }
 
     /// <summary>
-    /// The action representing the initiation of rating summary load action.
+    /// The action representing a failed dependent action.
+    /// </summary>
+    public class DependentFailAction : BaseFailAction
```

Comment:
Typo: In this action, `Dependent` should be `Dependents` to match the related actions.